src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -0,0 +1,161 @@
+#include "ray/core_worker/transport/direct_task_transport.h"
+
+namespace ray {
+
+void DoInlineObjectValue(const ObjectID &obj_id, std::shared_ptr<RayObject> value,
+                         const TaskSpecification &task) {
+  auto &msg = task.GetMutableMessage();
+  bool found = false;
+  for (size_t i = 0; i < task.NumArgs(); i++) {
+    auto count = task.ArgIdCount(i);
+    if (count > 0) {
+      const auto &id = task.ArgId(i, 0);
```

Comment:
Should handle both indices for `ArgId` here

Reply:
We RAY_CHECK that this isn't handled. I think this is java-specific.